asm/c128/stubs_test.go

```diff
@@ -0,0 +1,224 @@
+		x:   []complex128{1i, 1i, 1i},
+		y:   []complex128{1, 2, 1},
+		ex:  []complex128{2, 3, 2}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   -1i,
+		dst: []complex128{1i, 1i, 1i},
+		x:   []complex128{1i, 1i, 1i, 1i, 1i}[1:4],
+		y:   []complex128{1, 1, 2, 1, 1}[1:4],
+		ex:  []complex128{2, 3, 2}},
+	// Run big test twice, once aligned once unaligned.
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: 2, incY: 2, incDst: 3, ix: 0, iy: 0, idst: 0,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: -2, incY: -2, incDst: -3, ix: 18, iy: 18, idst: 27,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+	{incX: -2, incY: 2, incDst: -3, ix: 18, iy: 0, idst: 27,
+		a:   1 - 1i,
+		dst: make([]complex128, 10),
+		x:   []complex128{1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i, 1i},
+		y:   []complex128{1, 1, 2, 1, 1, 1, 1, 2, 1, 1},
+		ex:  []complex128{2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 2 + 1i, 3 + 1i, 2 + 1i, 2 + 1i}},
+}
+
+func guardVector(v []complex128, g complex128, g_ln int) (guarded []complex128) {
+	guarded = make([]complex128, len(v)+g_ln*2)
+	copy(guarded[g_ln:], v)
+	for i := 0; i < g_ln; i++ {
+		guarded[i] = g
+		guarded[len(guarded)-1-i] = g
+	}
+	return guarded
+}
+
+func validGuard(v []complex128, g complex128, g_ln int) bool {
+	for i := 0; i < g_ln; i++ {
+		if v[i] != g || v[len(v)-1-i] != g {
+			return false
+		}
+	}
+	return true
+}
+
+func TestAxpyUnitary(t *testing.T) {
+	for j, v := range tests {
+		g_ln := 4 + j%2
+		v.x, v.y = guardVector(v.x, 1, g_ln), guardVector(v.y, 1, g_ln)
+		x, y := v.x[g_ln:len(v.x)-g_ln], v.y[g_ln:len(v.y)-g_ln]
+		AxpyUnitary(v.a, x, y)
+		for i := range v.ex {
+			if y[i] != v.ex[i] {
+				t.Error("Test", j, "Unexpected result at", i, "Got:", y[i], "Expected:", v.ex[i])
+				t.Error(v.y)
+				t.FailNow()
+			}
+		}
+		if !validGuard(v.x, 1, g_ln) {
+			t.Error("Test", j, "Guard violated in x vector", v.x[:g_ln], v.x[len(v.x)-g_ln:])
+		}
+		if !validGuard(v.y, 1, g_ln) {
+			t.Error("Test", j, "Guard violated in y vector", v.y[:g_ln], v.y[len(v.x)-g_ln:])
+		}
+	}
+}
+
+func TestAxpyUnitaryTo(t *testing.T) {
+	for j, v := range tests {
+		g_ln := 4 + j%2
+		v.x, v.y = guardVector(v.x, 1, g_ln), guardVector(v.y, 1, g_ln)
+		v.dst = guardVector(v.dst, 0, g_ln)
```

Comment:
0 seems like a weird value for the guard. Could come from accidental zeroing, etc.